bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleCloudAPI.java

```diff
@@ -164,17 +173,50 @@ void setConfig(GoogleTTSConfig config) {
         }
     }
 
-    private Credentials createCredentials(String serviceAccountKey) throws IOException {
-        try (ByteArrayInputStream bis = new ByteArrayInputStream(serviceAccountKey.getBytes())) {
-            GoogleCredentials credential = GoogleCredentials.fromStream(bis)
-                    .createScoped(Collections.singleton(GCP_SCOPE));
-            return FixedCredentialsProvider.create(credential).getCredentials();
+    /**
+     * Fetches the OAuth2 tokens from Google Cloud Platform if the auth-code is set in the configuration. If successful
+     * the auth-code will be removed from the configuration.
+     *
+     * @return true if the auth-code was not set or if the auth-code was successfully used to get a new refresh and
+     *         access token
+     */
+    private boolean checkOnAuthCode() {
+        String authcode = config.authcode;
+        if (authcode != null && !authcode.isEmpty()) {
+            logger.debug("Trying to get access and refresh tokens.");
+            try {
+                oAuthService.getAccessTokenResponseByAuthorizationCode(authcode, GCP_REDIRECT_URI);
+
+                config.authcode = null;
+
+                Configuration serviceConfig = configAdmin.getConfiguration(GoogleTTSService.SERVICE_PID);
+                Dictionary<String, Object> configProperties = serviceConfig.getProperties();
+                if (configProperties == null) {
+                    configProperties = new Hashtable<>();
```

Comment:
If configProperties is bull iis there a reason to set PARAM_AUTHCODE anyway? Looking at the rest of the code it seem to handle the case it doesn't exist.

Reply:
Yes, that is right. In theory this will never happen. It is just a fallback. I will change it and throw an `AuthorizationException` instead.